### PR TITLE
Fixes to metrics docs

### PR DESCRIPTION
**Pull request by pschisa**
Based on my testing, setting Teleport in debug mode is not required to expose the metrics

~~~
# cat /etc/systemd/system/teleport.service
ExecStart=/usr/local/bin/teleport start  --config=/etc/teleport.yaml --diag-addr=http://172.31.36.239:3434 --pid-file=/run/teleport/teleport.pid

# cat /etc/teleport.yaml
teleport:
  log:
    severity: INFO

# curl http://172.31.36.239:3434/metrics | more
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP audit_failed_disk_monitoring Number of times disk monitoring failed.
# TYPE audit_failed_disk_monitoring counter
audit_failed_disk_monitoring 0
# HELP audit_failed_emit_events Number of times emitting audit event failed.
# TYPE audit_failed_emit_events counter
audit_failed_emit_events 0
# HELP audit_percentage_disk_space_used Percentage disk space used.
# TYPE audit_percentage_disk_space_used gauge
audit_percentage_disk_space_used 100
# HELP audit_server_open_files Number of open audit files
# TYPE audit_server_open_files gauge
audit_server_open_files 0
...

# teleport version
Teleport Enterprise v12.1.2 git:api/v12.1.2-0-gbfd4ff0 go1.20.
~~~

Also, curling the diag endpoint without the metrics part returns a 404 page not found which is a confusing way to validate it is working

~~~
# curl http://172.31.36.239:3434/
404 page not found
~~~

### CODE REVIEW

**Review by rosstimothy — CHANGES_REQUESTED** [review]
This is included in several diagnostics pages, some of which do actually require `--debug` in order to work(most notably collecting pprof profiles).

**Review by zmb3 — APPROVED** [review]

**Comment by pschisa** [comment]
hmmm so this might not be a simple change after all.  I'll check how the doc's team wants to proceed @ptgott @alexfornuto

**Comment by alexfornuto** [comment]
It looks to me like `docs/pages/management/diagnostics/profiles.mdx` is the only page that expects the `--debug` flag in this partial. I tested a solve locally where we can add the extra flag when needed using a partial variable, and if we're all happy with that I can push it to this branch.

**Review by alexfornuto — APPROVED** [review]

File: docs/pages/includes/diagnostics/diag-addr-prereqs-tabs.mdx [hunk 1]
```diff
@@ -32,6 +39,6 @@ them via:
     Verify that Teleport is now serving the diagnostics endpoint:
 
     ```code
-    $ curl http://127.0.0.1:3000
+    $ curl http://127.0.0.1:3000/metrics
```
[the remark this answers is not in the document]
Comment (pschisa) [reply docs/pages/includes/diagnostics/diag-addr-prereqs-tabs.mdx:42]: Is it better to note that the expectation is a 404 in that case?  Or is there a better endpoint to probe?  Would recommending a ps aux statement be more appropriate?

[the remark this answers is not in the document]
Comment (pschisa) [reply docs/pages/includes/diagnostics/diag-addr-prereqs-tabs.mdx:42]: I agree that would be better, returns a nice status OK

**Review by rosstimothy — APPROVED** [review]
🎉

**Comment by pschisa** [comment]
v10 and v11 did not get the debug added so not backport required

**Pull request merged**